### PR TITLE
#159965441 Users should be able to view and edit their profiles.

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?
```
Users should be able to view and edit their profiles
```

## Description of Task to be completed?

```
Users should be able to view their profile
Users should be able to view other users profile 
Users should be able to edit their profiles
Users should be able to change their profile image
```
## How should this be manually tested?
```
1. Clone the repository
2. Check out the branch view-edit-profile-159965441
3. Run the command `yarn test` and for coverage run `yarn test --coverage`
```

## What are the relevant pivotal tracker stories?
[#159965441](https://www.pivotaltracker.com/story/show/159965441)

## Screenshots if available to support PR.

View Profile 
screen shot 2018-10-16 at 18 23 03

Requested user profile not found
screen shot 2018-10-16 at 18 27 05

View Edit Profile 
screen shot 2018-10-16 at 18 23 25

Cloudinary file upload
screen shot 2018-10-16 at 18 24 42

## Questions:
```
Was this PR helpful ??? 
```

### CODE REVIEW

File: src/containers/Profile/UpdateProfile/actions.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const url = 'https://ah-leagueoflegends-staging.herokuapp.com/api/user/';
+  return (dispatch) => {
+    dispatch(updateUserProfile());
+    return axios.put(url,
+      profile,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization:
+                'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MTgsImV4cCI6MTU0MjIxMzc0MX0.RoqZTO7S4ySw1bOzzBtLuV4y6wB4ChQl4-oiqqYy6hM',
```
Comment (kevintumbo) [inline src/containers/Profile/UpdateProfile/actions.js]: why is the authorization token hardcoded?

**Review by kevintumbo — CHANGES_REQUESTED** [review]

File: src/containers/Articles/Article/largArticle.js [hunk 2]
```diff
@@ -23,13 +27,18 @@ class MainArticle extends Component {
...
     }
     return null;
-  }
+  };
 
   render() {
-    const { mainArticle, history } = this.props;
+    const { viewProfileReducer } = this.props;
+    const { payload } = viewProfileReducer;
+    const { mainArticle } = this.props;
```
Comment (kipropbrian) [inline src/containers/Articles/Article/largArticle.js]: I think you can merge `mainArticle` and `viewProfileReducer` together.

File: src/containers/Profile/UpdateProfile/actions.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+});
+
+export const updateUserProfileFailure = errors => ({
+  type: UPDATE_PROFILE_FAILURE,
+  errors,
+});
+
+// update user details
+const updateUser = (profile) => {
+  const url = 'https://ah-leagueoflegends-staging.herokuapp.com/api/user/';
```
Comment (kipropbrian) [inline src/containers/Profile/UpdateProfile/actions.js]: This should be an env var

File: src/components/Navbar/Navbar.js [hunk 4]
```diff
@@ -1,18 +1,43 @@
...
+    }
+  }
 
   logout = () => {
     localStorage.removeItem('access_token');
     localStorage.removeItem('isAuthenticated');
     window.location.reload(true);
+  };
+
+  handleprofileView() {
```
[the remark this answers is not in the document]
Comment (ghost) [reply src/components/Navbar/Navbar.js:42]: Function declaration does not use 'this'.

**Review by Quantum-35 — APPROVED** [review]

**Comment by Mutuba** [comment]
@rethes kindly solve the merge conflicts in the pr so that I can review.

**Review by DerKip — APPROVED** [review]

**Review by Teddykavooh — APPROVED** [review]

**Review by Derrickkip — APPROVED** [review]

File: src/components/Header/UserInfo.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    this.setState(prevState => ({ profileClicked: !prevState.profileClicked }))
+  );
+
+  publishClickedHandler = () => (
+    this.setState(prevState => ({ publishClicked: !prevState.publishClicked }))
+  );
+
+  render() {
+    const { publishHandler, viewProfileReducer, username } = this.props;
+    // const { publishClicked } = this.state;
```
Comment (kevintumbo) [inline src/components/Header/UserInfo.js]: remove commented out code

File: src/components/Header/UserInfo.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                {username}
+                {' '}
+              </p>
+              <div className="article-time--details" style={{ padding: 1 }}>
+                {profile.bio}
+              </div>
+            </div>
+            {/* <button type="button" onClick={this.publishClickedHandler} className="button">
+                    Ready to Publish?
+            </button> */}
```
Comment (kevintumbo) [inline src/components/Header/UserInfo.js]: remove commented out code

File: src/components/Navbar/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+                        </select>
+                      </div>
+
+                    </form>
+                  </li>
+                  {(isAuthenticated) ? (
+                    <span>
+                      <li><i className="material-icons">notifications_none</i></li>
+                      <li><i className="material-icons">bookmark_border</i></li>
+                      {/* onClick={this.handleprofileView} see profile */ }
```
Comment (kevintumbo) [inline src/components/Navbar/Navbar.js]: remove commented out code

File: src/components/Navbar/Navbar.js [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+                      <li>
+                        {/* <button className="profile-button" type="submit"> */}
```
Comment (kevintumbo) [inline src/components/Navbar/Navbar.js]: another one

File: src/components/Navbar/Navbar.js [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+                        <img role="image" className="small-navbar-profile" onClick={this.handleprofileView} src={image} alt={username} />
+                        {/* </button> */}
```
Comment (kevintumbo) [inline src/components/Navbar/Navbar.js]: another one

**Review by kevintumbo — CHANGES_REQUESTED** [review]

File: src/components/Navbar/Navbar.js [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+import fetchUserDetails from '../containers/Profile/ViewProfile/actions';
+
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.searchState = { searchText: '' };
+
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleChange = this.handleChange.bind(this);
```
Comment (kevintumbo) [inline src/components/Navbar/Navbar.js:20]: bind not necessary if you are using arrow functions

**Review by kevintumbo — CHANGES_REQUESTED** [review]

**Pull request merged**